app/views/govuk_publishing_components/components/docs/show_password.yml

```diff
@@ -0,0 +1,72 @@
+name: Show password input
+description: A password input field that allows the password to be shown.
+body: |
+  Adds a password reveal button to an input that toggles its type from password to text, revealing the password. Does not appear if JS is disabled.
+
+  Uses a visually hidden bit of text to inform screen reader users of the state of the input rather than announcing the content of the password input when toggled.
+accessibility_criteria: |
+  The component must:
+
+  * indicate to the user that the password has been shown or hidden
+
+  Inputs in the component must:
+
+  * accept focus
+  * be focusable with a keyboard
+  * be usable with a keyboard
+  * be usable with touch
+  * indicate when they have focus
+  * be recognisable as form input elements
+  * have correctly associated labels
+  * be of the appropriate type for their use, e.g. password inputs should be of type 'password'
+
+  Buttons in the component must:
+
+  - accept focus
+  - be focusable with a keyboard
+  - indicate when it has focus
+  - activate when focused and space is pressed
+  - activate when focused and enter is pressed
+govuk_frontend_components:
+  - text-input
+examples:
+  default:
+    data:
+      label:
+        text: Please enter your password
+      value: "this is my password"
+  disable_form_submit_check:
+    description: The component sets the `autocomplete` attribute on the input to "off" so that browsers don't remember the password. This isn't reliable, so the component reverts the input type back to a password when its parent form is submitted. If this is causing conflicts with other JS on your form, you can set `disable-form-submit-check` to `true` to disable this functionality and implement it yourself.
+    data:
+      label:
+        text: Please enter your password
+      value: "this is my password"
+      disable_form_submit_check: true
+  with_attributes:
+    description: The component accepts many but not all of the options that can be passed to a regular input, including `id`, `name` and `describedby` (not shown).
+    data:
+      label:
+        text: Please enter your password
+      value: "this is my password"
+      id: "custom_id"
+      name: "custom_name"
+  with_hint:
+    data:
+      label:
+        text: Please enter your password
+      value: "this is my password"
+      hint: "Your password must be at least twelve thousand characters"
+  with_error:
+    data:
+      label:
+        text: Please enter your password
+      value: "this is my password"
+      error_message: No it isn't
```

Comment:
Are you concerned at all that the show/reveal button lacks the error border? It didn't seem a big concern for me but thought I'd check.

Reply:
I don't think I'm concerned, no. If an error is present, it's to do with the contents of the input box, not the show/hide password button. However, I have pushed a new commit to address the slightly broken focus/active outline overlap problem.